kubernetes/redpanda/templates/statefulset.yaml

```diff
@@ -0,0 +1,66 @@
+apiVersion: apps/v1
+kind: StatefulSet
+metadata:
+  name: {{ template "redpanda.name" . }}
+spec:
+  selector:
+    matchLabels:
+      app: {{ template "redpanda.name" . }}
```

Comment:
Should this tie in with `redpanda.labels`? i.e., `app.kubernetes.io/name:  {{ template "redpanda.name" . }}` 

https://kubernetes.io/docs/concepts/overview/working-with-objects/common-labels/

Component should be added, e.g., `redpanda`, and then we might also have `coprocessor` or `pandaproxy`.

I've seen version required, possibly as part of Istio: https://istio.io/latest/docs/ops/deployment/requirements/#pod-requirements

Reply:
I probably wouldn't template these labels.

Reply:
I will make changes but do you really think we shouldn't template them at all? I was trying to follow the same approach as CockroachDB. 

https://github.com/cockroachdb/helm-charts/blob/master/cockroachdb/templates/statefulset.yaml

I'd like to get @rkruze's opinion too.

Reply:
@BenPope I revised the labelling scheme. I'm not sure if all of your concerns have been addressed. See what you think. I can do some more work on this if necessary.